server/src/test/java/org/elasticsearch/action/admin/cluster/snapshots/create/CreateSnapshotRequestTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package org.elasticsearch.action.admin.cluster.snapshots.create;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.elasticsearch.common.settings.Settings.readSettingsFromStream;
+import static org.elasticsearch.common.settings.Settings.writeSettingsToStream;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.tasks.TaskId;
+import org.junit.Test;
+
+import io.crate.common.unit.TimeValue;
+import io.crate.metadata.PartitionName;
+import io.crate.metadata.RelationName;
+
+public class CreateSnapshotRequestTest {
+
+    @Test
+    public void test_streaming_bwc_can_be_read_on_5() throws Exception {
+        RelationName table = new RelationName("foo", "bar");
+
+        CreateSnapshotRequest request = new CreateSnapshotRequest("my_repo", "my_snapshot");
+        request.relationNames(List.of(table));
+
+        try (var out = new BytesStreamOutput()) {
+            out.setVersion(Version.V_5_10_0);
+            request.writeTo(out);
+            try (var in = out.bytes().streamInput()) {
+                // parent request
+                TaskId.readFromStream(in);
+                in.readTimeValue();
+
+                // CreateSnapshotRequest
+                var snapshot = in.readString();
```

Comment:
Why did you choose this verbatim way, instead of creating `CreateSnapshotRequest` and then use 2 different versions in/out to test the whole roundtrip?

Reply:
Because the request has no indices or templates anymore.

Reply:
of course, sorry! :facepalm: